client/rest/src/test/java/org/elasticsearch/client/SyncResponseListenerTests.java

```diff
@@ -1,273 +0,0 @@
-/*
- * Licensed to Elasticsearch under one or more contributor
- * license agreements. See the NOTICE file distributed with
- * this work for additional information regarding copyright
- * ownership. Elasticsearch licenses this file to you under
- * the Apache License, Version 2.0 (the "License"); you may
- * not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-package org.elasticsearch.client;
-
-import org.apache.http.ConnectionClosedException;
-import org.apache.http.HttpHost;
-import org.apache.http.HttpResponse;
-import org.apache.http.ProtocolVersion;
-import org.apache.http.RequestLine;
-import org.apache.http.StatusLine;
-import org.apache.http.conn.ConnectTimeoutException;
-import org.apache.http.message.BasicHttpResponse;
-import org.apache.http.message.BasicRequestLine;
-import org.apache.http.message.BasicStatusLine;
-
-import java.io.IOException;
-import java.io.PrintWriter;
-import java.io.StringWriter;
-import java.net.SocketTimeoutException;
-import java.net.URISyntaxException;
-import javax.net.ssl.SSLHandshakeException;
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertSame;
-import static org.junit.Assert.fail;
-
-public class SyncResponseListenerTests extends RestClientTestCase {
```

Comment:
I think when you drop this file you lose the tests that `assertExceptionStackContainsCallingMethod` was so careful about. You preserve that test in `RestClientSingleHostTests` but only for `ResponseException`. This had the advantage of covering all of these other exception types. And I think it is important that we make sure the stack trace is right for them as well.

Reply:
It may well be right in your PR but I think we should have some tests that assert it. Basically I think these tests need to stay "somehow", but I'm not sure how given how you've changed the code.